### ISSUE TITLE
Running out of CUDA/GPU spaces

### ISSUE THREAD

**Issue by gmegh** [lucidrains/phenaki-pytorch#14]
I have a GPU with 15GB and it seems it runs out of space when I try to train the network with 50 videos at a time. Do you think it would be better to repeat the loss training video per video, instead of all the videos at once?

**Comment by gmegh** [comment]
Additionally, when training on 20 videos and text prompts the model output is still just noise, which I think is the expected result, given the lack of training, right?

**Comment by gmegh** [comment]
Yes, that makes, sense. Let me know if I can help. Do you know when are you planning on having the agnostic feature ready?

I did create some short functions to be able to use .mp4 instead of just gifs and saved the tensors to mp4 as well. Let me know if you would like for me to add them to a PR

**Comment by lucidrains** [comment]
@gmegh so i have to add 3d continuous relative positional bias to the maskgit embedding to allow for generalization to different sizes. i think i should be able to get it done by tomorrow evening

re: mp4 - yes! that would be super helpful!

**Comment by gmegh** [comment]
Great! I will create a PR. 

Also for reference, these guys are also working on implementing it: https://github.com/LAION-AI/phenaki

I think another nice to-do would be to allow for saving the trained model and be able to load it

**Comment by lucidrains** [comment]
@gmegh yup, i've been chatting with Dominic

they are planning on straying a bit farther from the paper's implementation (for example, using all convolutions in the cvivit)

but this is a joint effort; anything i develop here they are free to use

**Comment by lucidrains** [comment]
@gmegh yea, i'll definitely get to the training code soon, once i add a few more bells and whistles to the attention networks

**Comment by gmegh** [comment]
Awesome! Happy to help if you want.

**Comment by lucidrains** [comment]
@gmegh yea definitely welcome any help!

do you know of any good packages for processing and loading video data?

**Comment by gmegh** [comment]
Yes! I think cv2 is a good package. I made some quick functions with it that I have added to the new PR. The crop_image() should probably be edited further

**Comment by gmegh** [comment]
What is the status of the code right now? I think the checkboxes in the readme are outdated, right?

**Comment by lucidrains** [comment]
@gmegh feel free to experiment in the mean time!

**Comment by gmegh** [comment]
Hi @lucidrains ! Is the framework that can produce unconditional (or text conditioned) images ready? I am experimenting with the current version and I would need a way to train by batches, because using 500 videos at a time already fills up my CUDA memory. Any idea on how to go about this?

**Comment by cyrilzakka** [comment]
> @gmegh yea definitely welcome any help!
> 
> do you know of any good packages for processing and loading video data?

I could take care of this. Any preferences as to whether you'd like to break down each video into frames, or sample from a video directly?
